modules/msk/polysolver/main.nf

```diff
@@ -0,0 +1,62 @@
+process POLYSOLVER {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://sachet/polysolver:v3':
+        'docker.io/sachet/polysolver:v3' }"
+
+    input:
+    tuple val(meta), path(bam)
+    val(includeFreq)
+    val(build)
+    val(format)
+    val(insertCalc)
+
+    output:
+    tuple val(meta), path("*.hla.txt"), emit: hla
+    path  "versions.yml"              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args              = task.ext.args ?: ''
+    def prefix            = task.ext.prefix ?: "${meta.id}"
+    def includeFreq_param = includeFreq ?: 1
+    def insertCalc_param  = insertCalc ?: 0
+    def build_param       = build ? (["GRCh37","hg19"].contains(build) ? "hg19" : "hg38") : "hg19"
```

Comment:
I'm a little unfamiliar with the tool, but are all of these parameters required? If any are optional, could they be removed and specified with`args` if needed? All good to ignore if these are all required / best to be specified directly.

Reply:
they are not optional and unfortunately they're positional...here's an example:
```
bash shell_call_hla_type \
    sample.bam \
    Unknown \
    1 \
    hg19 \
    STDFQ \
    0 \
    output/
```
i kind of struggled to write something that was easy and straightforward for a user and kind of settled on making the args explicit in the input channels, but i'm not really convinced by it either. do you think it makes more sense to write:
```
def args = task.ext.args ?: "Unknown 1 hg19 STDFQ 0"
"""
bash shell_call_hla_type \
    sample.bam \
    $args \
    ${prefix}/
"""
```
?

Reply:
Ah I see. In that case, I think the first way, the way it's currently implemented, is best since there really are only positional arguments.